standalone/levenshtein.cpp

```diff
@@ -9,15 +9,30 @@
 #include <edlib.h>
 #include <fstream>
 
+namespace {
+    struct AlignResult {
+        EdlibAlignResult self;
+        AlignResult(std::string_view s1, std::string_view s2) {
+            self = edlibAlign(
+                s1.data(),
+                s1.size(),
+                s2.data(),
+                s2.size(),
+                edlibDefaultAlignConfig());
+        }
+        ~AlignResult() noexcept {
+            edlibFreeAlignResult(self);
+        }
+        std::size_t
+        editDistance() {
+            return self.editDistance;
+        }
+    };
+} // namespace
+
 std::size_t
 levenshtein_distance(std::string_view s1, std::string_view s2) {
-    return edlibAlign(
-               s1.data(),
-               s1.size(),
-               s2.data(),
-               s2.size(),
-               edlibDefaultAlignConfig())
-        .editDistance;
+    return AlignResult{ s1, s2 }.editDistance();
```

Comment:
The semantics of `AlignResult` seems a little weird: `self` is public but `self.editDistance` is accessed through `editDistance()`, the operation takes place in the constructor of a result type, and so on...

Since the struct won't be reused, why not just call the canonical provided API directly here and call it a day:

```cpp
std::size_t
levenshtein_distance(std::string_view s1, std::string_view s2) {
    auto r = edlibAlign(
               s1.data(),
               s1.size(),
               s2.data(),
               s2.size(),
               edlibDefaultAlignConfig());
    std::size_t d = r.editDistance;
    edlibFreeAlignResult(r);
    return d;
}
```

Reply:
I was unsure about exceptions, but looking around in edlib source code it seems more likely to get a SIGSEGV than an exception. And your comment on semantics is relevant. I'll fix that.

Reply:
> I was unsure about exceptions, 

No exceptions in C :)

> but looking around in edlib source code it seems more likely to get a SIGSEGV than an exception. And your comment on semantics is relevant. I'll fix that.

Great. Did you check if the new version also fixes the original problem locally for you?

Reply:
> > I was unsure about exceptions,
> 
> No exceptions in C :)

Indeed, we have `extern "C"` in the header. We also have `include <vector>` and what-not-else in the implementation file (which is cpp, ofc) but not a single `catch` clause anywhere. If any piece of implementation code throws an exception all bets are off. 😅 
Maybe I should take some time to learn the algorithms and propose some changes in edlib.

> > but looking around in edlib source code it seems more likely to get a SIGSEGV than an exception. And your comment on semantics is relevant. I'll fix that.
> 
> Great. Did you check if the new version also fixes the original problem locally for you?

Yeap! No mention to edlib in the memory leak section. The motivating issue is fixed.

![image](https://user-images.githubusercontent.com/11138291/221684789-ce2c6ef1-9436-46cf-b74f-c39d89042ee9.png)